test/models/user_test.rb

```diff
@@ -3,7 +3,19 @@
 require 'test_helper'
 
 class UserTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test '#name_or_email should return name if user has name' do
+    user = FactoryBot.build(:user)
+    user.name = 'Alice'
+    user.email = 'alice@example.com'
```

Comment:
```suggestion
    user = FactoryBot.build(:user, name: 'Alice', email: 'alice@example.com')
```

のようにしてデフォルト値をオーバーライドするといいと思います。

Reply:
修正いたしました。